src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/XmlSerializerGenerator.cs

```diff
@@ -124,21 +122,142 @@ private List<Type> CollectXmlSerializerTypes(Assembly assembly, List<XmlMapping>
+                throw new ToolRuntimeException();
+            }
+
+            Type xmlSerializerOperationBehaviorType = Tool.SMAssembly.GetType("System.ServiceModel.Description.XmlSerializerOperationBehavior");
+            if (xmlSerializerOperationBehaviorType == null)
+            {
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.XmlSerializerOperationBehaviorType in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            Type operationType = Tool.SMAssembly.GetType("System.ServiceModel.Description.OperationDescription");
+            if (operationType == null)
+            {
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.OperationDescription in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            PropertyInfo getBehaviors = operationType.GetProperty("Behaviors");
+            if (getBehaviors == null)
+            {
+                ToolConsole.WriteError($"Not found method get_Behaviors in type {operationType}");
+                throw new ToolRuntimeException();
+            }
+
+            Type KeyedByTypeCollection = Tool.SMAssembly.GetType("System.Collections.Generic.KeyedByTypeCollection`1");
+            if (KeyedByTypeCollection == null)
+            {
+                ToolConsole.WriteError($"Not found type System.Collections.Generic.KeyedByTypeCollection`1 in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            Type IOperationBehavior = Tool.SMAssembly.GetType("System.ServiceModel.Description.IOperationBehavior");
+            if (IOperationBehavior == null)
+            {
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.IOperationBehavior in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            KeyedByTypeCollection = KeyedByTypeCollection.MakeGenericType(new Type[] { IOperationBehavior });
+            if (KeyedByTypeCollection == null)
+            {
+                ToolConsole.WriteError($"Cannot make Generic Type System.Collections.Generic.KeyedByTypeCollection<IOperationBehavior> in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            MethodInfo find = KeyedByTypeCollection.GetMethod("Find");
+            if(find == null)
+            {
+                ToolConsole.WriteError($"Not found method find in type {KeyedByTypeCollection}");
+                throw new ToolRuntimeException();
+            }
+
+            MethodInfo findXmlSerializerOperationBehavior = find.MakeGenericMethod(new Type[] { xmlSerializerOperationBehaviorType });
+            if (findXmlSerializerOperationBehavior == null)
```

Comment:
There's no need for a new variable to hold this. You can reuse the same one like you did for KeyedByTypeCollection